wgpu/src/backend/web.rs

```diff
@@ -1151,12 +1151,23 @@ impl crate::Context for Context {
         ]
     }
 
+    fn surface_get_supported_modes(
+        &self,
+        _surface: &Self::SurfaceId,
+        _adapter: &Self::AdapterId,
+    ) -> Vec<wgt::PresentMode> {
+        vec![wgt::PresentMode::Fifo]
+    }
+
     fn surface_configure(
         &self,
         surface: &Self::SurfaceId,
         device: &Self::DeviceId,
         config: &wgt::SurfaceConfiguration,
     ) {
+        if let wgt::PresentMode::Mailbox | wgt::PresentMode::Immediate = config.present_mode {
+            panic!("Only FIFO/Auto* is supported on web");
+        }
```

Comment:
What do you think of making `Surface::configure` return an error instead of panicking? Like `UnsupportedPresentMode`.

Reply:
I'd hold off on it for now, as it gets into similar problems from #2757 where some errors come in a result, some come asynchronously and it gets complicated to communicate this to the user. I'm not generally opposed, just don't want to tackle it in this PR.

Reply:
Sounds good!